plugins/obs-ffmpeg/obs-ffmpeg-source.c

```diff
@@ -463,26 +472,40 @@ static void ffmpeg_source_update(void *data, obs_data_t *settings)
 
 	stop_reconnect_thread(s);
 
+	// Restart media source if these properties are changed
+	is_hw_decoding = obs_data_get_bool(settings, "hw_decode");
+	should_restart_media |= (s->is_hw_decoding != is_hw_decoding);
+	range = obs_data_get_int(settings, "color_range");
+	should_restart_media |= (s->range != range);
+	speed_percent = (int)obs_data_get_int(settings, "speed_percent");
+	if (speed_percent < 1 || speed_percent > 200)
+		speed_percent = 100;
+	should_restart_media |= (s->speed_percent != speed_percent);
 	ffmpeg_options = obs_data_get_string(settings, "ffmpeg_options");
+	if (s->ffmpeg_options && !should_restart_media)
+		should_restart_media |=
+			strcmp(s->ffmpeg_options, ffmpeg_options) != 0;
```

Comment:
I'd personally prefer something like:

```cpp
	is_hw_decoding = obs_data_get_bool(settings, "hw_decode");
	range = obs_data_get_int(settings, "color_range");
	speed_percent = (int)obs_data_get_int(settings, "speed_percent");
	if (speed_percent < 1 || speed_percent > 200)
		speed_percent = 100;
	ffmpeg_options = obs_data_get_string(settings, "ffmpeg_options");

	/* Restart media source if these properties are changed */
	if (s->is_hw_decoding != is_hw_decoding || s->range != range ||
	    s->speed_percent != speed_percent ||
	    (s->ffmpeg_options &&
	     strcmp(s->ffmpeg_options, ffmpeg_options) != 0))
		should_restart_media = true;
```
It's not that much cleaner, but it explicitly separates variable assignment and the check to restart media into their own chunks of code. Also would prefer `/* */` style comments (more info on that in the next comment)